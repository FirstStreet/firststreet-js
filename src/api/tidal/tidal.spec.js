const FirstStreet = require('../../__mocks__/FirstStreet');
const tidalMock = require('../../__mocks__/tidalProperty');

const validMethods = ['getPropertyByFSID', 'getCityByFSID', 'getPropertyByLatLng', 'getCityByLatLng', 'getPropertyByAddress', 'getCityByAddress'];

describe('tidal', () => {
  const fsf = new FirstStreet('aa.bb.cc', {
    host: 'https://FAKE-HOST-FOR-TESTING.com',
  ***REMOVED***);

  it('should contain valid methods', () => {
    validMethods.forEach((method) => {
      expect(fsf.tidal[method]).toBeDefined();
    ***REMOVED***);
  ***REMOVED***);

  it('.getPropertyByFSID should get property by ID', async () => {
    const property = await fsf.tidal.getPropertyByFSID(tidalMock.FSID);

    expect(property).toMatchSnapshot();
    expect(property.results[0].floodType).toBe(tidalMock.results[0].floodType);
  ***REMOVED***);

  it('.getPropertyByLatLng should get property by lat, lng', async () => {
    const property = await fsf.tidal.getPropertyByLatLng(-75.6451595133313, -75.6451595133313);

    expect(property).toMatchSnapshot();
    expect(property.results[0].floodType).toBe(tidalMock.results[0].floodType);
  ***REMOVED***);

  it('.getPropertyByAddress should get property by address lookup', async () => {
    const property = await fsf.tidal.getPropertyByAddress('212 appoquin s, middletown, delware');

    expect(property).toMatchSnapshot();
    expect(property.results[0].floodType).toBe(tidalMock.results[0].floodType);
  ***REMOVED***);
***REMOVED***);
