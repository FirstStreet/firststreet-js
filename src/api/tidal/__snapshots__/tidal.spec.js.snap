// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tidal .getPropertyByAddress should get property by address lookup 1`] = `
Response {
  "data": Object {
    "FSID": 450350219571,
    "results": Array [
      Object {
        "data": Array [
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.09542,
              ***REMOVED***,
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.05723,
              ***REMOVED***,
            ],
            "year": 2018,
          ***REMOVED***,
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.09924,
              ***REMOVED***,
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.0619,
              ***REMOVED***,
            ],
            "year": 2023,
          ***REMOVED***,
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.12977,
              ***REMOVED***,
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.06892,
              ***REMOVED***,
            ],
            "year": 2028,
          ***REMOVED***,
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.14885,
              ***REMOVED***,
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.07514,
              ***REMOVED***,
            ],
            "year": 2033,
          ***REMOVED***,
        ],
        "floodID": "em",
        "floodType": "tidal flooding",
      ***REMOVED***,
      Object {
        "data": Array [
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.05725,
              ***REMOVED***,
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.02527,
              ***REMOVED***,
            ],
            "year": 2018,
          ***REMOVED***,
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.05725,
              ***REMOVED***,
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.02921,
              ***REMOVED***,
            ],
            "year": 2023,
          ***REMOVED***,
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.0687,
              ***REMOVED***,
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.03647,
              ***REMOVED***,
            ],
            "year": 2028,
          ***REMOVED***,
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.08015,
              ***REMOVED***,
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.04438,
              ***REMOVED***,
            ],
            "year": 2033,
          ***REMOVED***,
        ],
        "floodID": "kt",
        "floodType": "tidal flooding",
      ***REMOVED***,
    ],
  ***REMOVED***,
***REMOVED***
`;

exports[`tidal .getPropertyByFSID should get property by ID 1`] = `
Response {
  "data": Object {
    "FSID": 450350219571,
    "results": Array [
      Object {
        "data": Array [
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.09542,
              ***REMOVED***,
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.05723,
              ***REMOVED***,
            ],
            "year": 2018,
          ***REMOVED***,
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.09924,
              ***REMOVED***,
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.0619,
              ***REMOVED***,
            ],
            "year": 2023,
          ***REMOVED***,
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.12977,
              ***REMOVED***,
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.06892,
              ***REMOVED***,
            ],
            "year": 2028,
          ***REMOVED***,
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.14885,
              ***REMOVED***,
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.07514,
              ***REMOVED***,
            ],
            "year": 2033,
          ***REMOVED***,
        ],
        "floodID": "em",
        "floodType": "tidal flooding",
      ***REMOVED***,
      Object {
        "data": Array [
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.05725,
              ***REMOVED***,
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.02527,
              ***REMOVED***,
            ],
            "year": 2018,
          ***REMOVED***,
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.05725,
              ***REMOVED***,
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.02921,
              ***REMOVED***,
            ],
            "year": 2023,
          ***REMOVED***,
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.0687,
              ***REMOVED***,
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.03647,
              ***REMOVED***,
            ],
            "year": 2028,
          ***REMOVED***,
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.08015,
              ***REMOVED***,
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.04438,
              ***REMOVED***,
            ],
            "year": 2033,
          ***REMOVED***,
        ],
        "floodID": "kt",
        "floodType": "tidal flooding",
      ***REMOVED***,
    ],
  ***REMOVED***,
***REMOVED***
`;

exports[`tidal .getPropertyByLatLng should get property by lat, lng 1`] = `
Response {
  "data": Object {
    "FSID": 450350219571,
    "results": Array [
      Object {
        "data": Array [
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.09542,
              ***REMOVED***,
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.05723,
              ***REMOVED***,
            ],
            "year": 2018,
          ***REMOVED***,
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.09924,
              ***REMOVED***,
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.0619,
              ***REMOVED***,
            ],
            "year": 2023,
          ***REMOVED***,
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.12977,
              ***REMOVED***,
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.06892,
              ***REMOVED***,
            ],
            "year": 2028,
          ***REMOVED***,
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.14885,
              ***REMOVED***,
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.07514,
              ***REMOVED***,
            ],
            "year": 2033,
          ***REMOVED***,
        ],
        "floodID": "em",
        "floodType": "tidal flooding",
      ***REMOVED***,
      Object {
        "data": Array [
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.05725,
              ***REMOVED***,
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.02527,
              ***REMOVED***,
            ],
            "year": 2018,
          ***REMOVED***,
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.05725,
              ***REMOVED***,
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.02921,
              ***REMOVED***,
            ],
            "year": 2023,
          ***REMOVED***,
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.0687,
              ***REMOVED***,
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.03647,
              ***REMOVED***,
            ],
            "year": 2028,
          ***REMOVED***,
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              ***REMOVED***,
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.08015,
              ***REMOVED***,
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.04438,
              ***REMOVED***,
            ],
            "year": 2033,
          ***REMOVED***,
        ],
        "floodID": "kt",
        "floodType": "tidal flooding",
      ***REMOVED***,
    ],
  ***REMOVED***,
***REMOVED***
`;
